lib/datadog/core/configuration/components.rb

```diff
@@ -8,18 +8,18 @@
 require_relative '../telemetry/client'
 require_relative '../workers/runtime_metrics'
 
-require_relative '../../tracing/tracer'
-require_relative '../../tracing/flush'
-require_relative '../../tracing/sync_writer'
-require_relative '../../tracing/sampling/span/rule_parser'
-require_relative '../../tracing/sampling/span/sampler'
+require_relative '../../tracing/component'
+require_relative '../../profiling/component'
 
 module Datadog
   module Core
     module Configuration
       # Global components for the trace library.
       class Components
         class << self
+          include Datadog::Tracing::Component
+          include Datadog::Profiling::Component
```

Comment:
So, I know you're going for the smallest possible change here, and I can live with this one, buuuuuut I'm going to try to convince you to go a tiny small step further.

Can I convince you to, instead of re-injecting the original behaviors back using a module, to modify the `initialize` method below to instead call directly into the modules?

E.g. instead of 
```ruby
        def initialize(settings)
          # Logger
          @logger = self.class.build_logger(settings)

          agent_settings = AgentSettingsResolver.call(settings, logger: @logger)

          # Tracer
          @tracer = self.class.build_tracer(settings, agent_settings)

          # Profiler
          @profiler = self.class.build_profiler(settings, agent_settings, @tracer)

          # ... etc...
```

Do:
 
```ruby
        def initialize(settings)
          # Logger
          @logger = self.class.build_logger(settings)

          agent_settings = AgentSettingsResolver.call(settings, logger: @logger)

          # Tracer
          @tracer = Datadog::Tracing::Component.build_tracer(settings, agent_settings)

          # Profiler
          @profiler = Datadog::Profiling::Component.build_profiler(settings, agent_settings, @tracer)

          # ... etc...
```

I like the new version better because it makes the dependency explicit, rather than implicit, and I think it moves us closer to a future where the components would register themselves somewhere and provide these functions to be called.

Reply:
> it makes the dependency explicit, rather than implicit

I agree, and this explicit dependencing will also fail (which is a good thing!) if there's a reference in these methods to a ivar or method of `Core::Configuration::Components`... sadly this needs further validation than I can afford right now, hence this first noop move whose goal is to tackle source code ownership.

I'll attempt to tackle live code dependency in a second PR.